e2e/pages/charts/product/product.readonly.spec.js

```diff
@@ -16,6 +16,9 @@ describe('on charts page - Unique product chart', () => {
 
   it('should only show 2015 edition products', () => {
     browser.waitUntil(() => page.chartTitle.isDisplayed(), config.shortTimeout);
+    if (!page.chartTitle.isDisplayed()) {
+      browser.refresh();//stale element exception needs page refresh
```

Comment:
It worries me seeing a page refresh command in any of our tests. With our SPA app, the users never refresh (or shouldn't need to) so our tests shouldn't either. Is there some other way of making this change happen? Maybe navigate to a different chart and back or toggle the filter at the bottom of the chart?

Reply:
I agree completely. But I tried with all options suggested above. In first try, this test fails due to stale element and second test it passes. To have to pass, refresh is needed.

Reply:
I checked out this branch and ran it from scratch, and it failed. I ran it again after putting `takeScreenshot` commands in to see what was happening, and it passed. I took those out, removed this line, and it failed, then ran it again without changing anything and it passed

I'm not going to put this line of code into the final PR, because I don't think it should be there. We'll need to find another solution, but I'm going to work on that now, as this main PR is closest to going to getting deployed, so will merge this PR and see what I can find

Reply:
Yes this exception is not constantly happening. When I ran in local, it does passes the second time of retry if it fails the first time.